src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -191,7 +191,7 @@
      * The access control context of the creating thread.
      * Will be set to null when the MBean is not registered.
      */
-    private final AccessControlContext acc;
+    private volatile AccessControlContext acc;
```

Comment:
Is volatile necessary given that it's only updated via `completeTermination`?

Reply:
Yes I think it's better to make it `volatile`. It certainly doesn't hurt.